src/clients/vendor.rb

```diff
@@ -98,18 +98,16 @@ def main
       else
         Pkg.TargetInit("/", false)
 
-        @products = Pkg.ResolvableProperties("", :product, "")
+        @products = Y2Packager::Resolvable.find(kind: :product,
+           status:    :installed,
+           category:  "base")
         @products = [] if @products == nil
-        @products = Builtins.filter(@products) do |p|
-          Ops.get(p, "status") == :installed
-        end
-        @base = Builtins.filter(@products) do |p|
-          Ops.get_string(p, "category", "") == "base"
-        end
         @base = deep_copy(@products) if Builtins.size(@base) == 0
-        @product = Ops.get(@base, 0, {})
-        @version = Ops.get_string(@product, "version", "")
-        @version = Ops.get(Builtins.splitstring(@version, "-"), 0, "") # split off release
+        @product = @base[0]
+        if @product
+          @version = @product.version || ""
+          @version = Ops.get(Builtins.splitstring(@version, "-"), 0, "") # split off release
```

Comment:
You can use the `"version-version"` attribute (or the `.version-version` method), that contains the version without the release suffix. (For the backward compatibility I could not change the "version" result, I had to add a new attribute.)

I think just using simple
```suggestion
          @version = @product.version_version
```
should be enough.